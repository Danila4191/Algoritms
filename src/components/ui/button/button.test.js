import renderer from "react-test-renderer";
import  {render, screen, fireEvent}  from "@testing-library/react";
import { Button } from "./button";
//кнопка
it("рендер кнопки с текстом",() => {
  const tree = renderer
  .create(<Button text="Добавить" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
})

it("рендер кнопки без текста",() => {
  const tree = renderer
  .create(<Button />)
  .toJSON();
  expect(tree).toMatchSnapshot();
})
it("рендер кнопки заблокированной",() => {
  const tree = renderer
  .create(<Button disabled={true}/>)
  .toJSON();
  expect(tree).toMatchSnapshot();
})
it("рендер кнопки с лоадером",() => {
  const tree = renderer
  .create(<Button loading={true}/>)
  .toJSON();
  expect(tree).toMatchSnapshot();
})
it("рендер кнопки вызов колбека",() => {
  window.alert = jest.fn();
  render(<Button text="Добавить" onClick={()=>alert("добавили")}/>)
  const button = screen.getByText("Добавить");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith('добавили');
})