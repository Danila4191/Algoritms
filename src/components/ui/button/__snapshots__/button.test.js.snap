// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`рендер кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`рендер кнопки заблокированной 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`рендер кнопки с лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  loading={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`рендер кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Добавить
  </p>
</button>
`;
